From akpm@linux-foundation.org  Fri Apr 18 14:55:07 2008
From: Adrian Bunk <bunk@kernel.org>
Date: Fri, 18 Apr 2008 13:53:55 -0700
Subject: PCI: pci_scan_device() mustn't be __devinit
To: greg@kroah.com
Cc: linux-pci@atrey.karlin.mff.cuni.cz, akpm@linux-foundation.org, bunk@kernel.org
Message-ID: <200804182053.m3IKru3m015135@imap1.linux-foundation.org>


From: Adrian Bunk <bunk@kernel.org>

WARNING: drivers/pci/built-in.o(.text+0x150f): Section mismatch in reference from the function pci_scan_single_device() to the function .devinit.text:pci_scan_device()

Signed-off-by: Adrian Bunk <bunk@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/probe.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -875,8 +875,7 @@ EXPORT_SYMBOL(alloc_pci_dev);
  * Read the config data for a PCI device, sanity-check it
  * and fill in the dev structure...
  */
-static struct pci_dev * __devinit
-pci_scan_device(struct pci_bus *bus, int devfn)
+static struct pci_dev *pci_scan_device(struct pci_bus *bus, int devfn)
 {
 	struct pci_dev *dev;
 	u32 l;
